pkg/private/archive.py

```diff
@@ -174,11 +174,14 @@ def __init__(self,
                                               stdout=open(name, 'wb'))
       self.fileobj = self.compressor_proc.stdin
     self.name = name
-    self.root_directory = root_directory.rstrip('/').rstrip('\\')
-    self.root_directory = self.root_directory.replace('\\', '/')
+    # tarfile uses / instead of os.path.sep
+    self.root_directory = root_directory.replace(os.path.sep, '/').rstrip('/')
+    if self.root_directory:
+      self.root_directory = self.root_directory + '/'
 
     self.tar = tarfile.open(name=name, mode=mode, fileobj=self.fileobj)
     self.members = set([])
+    # The directories we have created so far
     self.directories = set([])
```

Comment:
Nit: just saying `set` should be vaild and still produce the same result as this.

Reply:
Yes. Changed.